### PR TITLE
appsync-data-with-images

### PR DESCRIPTION
**Pull request by mtliendo**

### CODE REVIEW

File: appsync-data-with-images/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# AWS Service 1 to AWS Service 2
+
+This pattern << explain usage >>
+
```
Comment (julianwood) [inline appsync-data-with-images/README.md:4]: Looks like this README isn't updated for this pattern with deployment instructions

File: appsync-data-with-images/cdk/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Welcome to your CDK TypeScript project
```
Comment (julianwood) [inline appsync-data-with-images/cdk/README.md:1]: Can this file be removed?

**Comment by julianwood** [comment]
Looks like folder is generic. Suggest renaming to `appsync-cognito-dynamodb-cdk` or something.
README doesn't include deployment instructions.

**Comment by mtliendo** [comment]
Thanks Julian! Carved out time tomorrow morning to get all these addressed, much appreciated!

**Comment by mtliendo** [comment]
> Looks like folder is generic. Suggest renaming to appsync-cognito-dynamodb-cdk or something.

Just double checking that you're referring to the `cdk` folder. Can definitely change, but looking at other repos, using `cdk` seems to be the standard.

File: appsync-data-with-images/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+   ```
+   git clone https://github.com/aws-samples/serverless-patterns
+   ```
+1. Change directory to the pattern directory:
+   ```
+   cd serverless-patterns/appsync-data-with-images
```
Comment (julianwood) [inline appsync-data-with-images/README.md:34]: Should this be cd serverless-patterns/appsync-data-with-images/cdk ?

Comment (julianwood) [inline appsync-data-with-images/README.md:34]: Think this should be: `cd serverless-patterns/appsync-data-with-images/cdk`

File: appsync-data-with-images/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+   ```
+1. Install dependencies
+   ```
+   npm install
+   ```
+1. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+   ```
+   npx aws-cdk deploy
```
Comment (julianwood) [inline appsync-data-with-images/README.md:42]: Think this should be `npx aws-cdk deploy --all`

File: appsync-data-with-images/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+   ```
+
+## Testing
+
+The easiest way to test the AppSync API is with the AppSync console at https://us-west-2.console.aws.amazon.com/appsync/home (change to your appropriate region)
+
```
Comment (julianwood) [inline appsync-data-with-images/README.md:48]: Is there any specific test someone could run to check this pattern is working?

File: appsync-data-with-images/cdk/bin/cdk.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
+import 'source-map-support/register'
+import * as cdk from 'aws-cdk-lib'
+import { AuthStack } from '../lib/AuthStack'
+import { FileStorageStack } from '../lib/fileStorageStack'
```
Comment (julianwood) [inline appsync-data-with-images/cdk/bin/cdk.ts:5]: Typo I think `FileStorageStack`

File: appsync-data-with-images/example-pattern.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+				"link": "https://aws.amazon.com/blogs/mobile/managing-images-in-your-nextjs-app-with-aws-appsync-and-the-aws-cdk/"
+			},
+			{
+				"text": "Amazon AppSync - Managed GraphQL service",
+				"link": "https://aws.amazon.com/appsync/"
+			}
+		]
+	},
+	"deploy": {
+		"text": ["<code>npx aws-cdk deploy</code>"]
```
Comment (julianwood) [inline appsync-data-with-images/example-pattern.json:36]: Add `--all`

File: appsync-data-with-images/example-pattern.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+		"headline": "How it works",
+		"text": [
+			"This sample project demonstrates how to use AWS AppSync to read and write to an Amazon DynamoDB table. The requests are protected with both Amazon Cognito userpools and identity pools. This pattern is leveraging the native integration between AppSync and DynamoDB to map the request and response.",
+			"With AppSync you can write a JavaScript function to contain your business logic. This function is contained within a pipeline resolver that associates the GraphQL schema with the datasource (DynamoDB in this case).",
+			"At a high-level, this pattern deploys one AppSync API, a Cognito Userpool and Identity pool, and a DynamoDB table."
+		]
+	},
+	"gitHub": {
+		"template": {
+			"repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/appsync-cognito-dynamodb",
```
Comment (julianwood) [inline appsync-data-with-images/example-pattern.json:17]: Is `appsync-cognito-dynamodb` the pattern name or `appsync-data-with-images`

File: appsync-data-with-images/example-pattern.json [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+		]
+	},
+	"deploy": {
+		"text": ["<code>npx aws-cdk deploy</code>"]
+	},
+	"testing": {
+		"text": ["See the Github repo for detailed testing instructions."]
+	},
+	"cleanup": {
+		"text": ["Delete the stack: <code>npx aws-cdk destroy</code>."]
```
Comment (julianwood) [inline appsync-data-with-images/example-pattern.json:42]: add `--all`

File: appsync-data-with-images/README.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+## Cleanup
+
+Run the given command to delete the resources that were created. It might take some time for the CloudFormation stack to get deleted
+
+```
+npx aws-cdk destroy
```
Comment (julianwood) [inline appsync-data-with-images/README.md:54]: This should be `npx aws-cdk destroy --all`

**Comment by julianwood** [comment]
> Looks like folder is generic. Suggest renaming to `appsync-cognito-dynamodb-cdk` or something. README doesn't include deployment instructions.

I think the pattern name should be `appsync-cognito-dynamodb-cdk` rather than `appsync-data-with-images`.
This isn't the `cdk` folder but the whole pattern folder. The pattern folder should be the same as in the `example-pattern.json`

**Comment by julianwood** [comment]
Hi @mtliendo Any update on this?

**Comment by jbesw** [comment]
Closing due to inactivity.

**Pull request closed without merging**